File: backend/app/schemas/payment.py
"""Payment (Razorpay) request/response schemas."""

from pydantic import BaseModel, Field


class PaymentOrderResponse(BaseModel):
    order_id: str
    amount: int = Field(..., description="Amount in smallest currency unit (paise for INR)")
    currency: str


class PaymentVerifyRequest(BaseModel):
    razorpay_order_id: str
    razorpay_payment_id: str
    razorpay_signature: str


class PaymentVerifyResponse(BaseModel):
    success: bool = True


class PaymentConfigResponse(BaseModel):
    """Public pricing hint for checkout UI (no secrets)."""

    price: int = Field(..., description="Active premium price in whole rupees")
    mode: str = Field(..., description='Effective pricing mode: "test" or "prod"')

File: frontend/src/lib/api/payment.ts
import { apiClient } from '@/lib/api/client'

export type PaymentConfigResponse = {
  price: number
  mode: 'test' | 'prod' | string
}

export type PaymentCreateOrderResponse = {
  order_id: string
  amount: number
  currency: string
}

export type PaymentVerifyBody = {
  razorpay_order_id: string
  razorpay_payment_id: string
  razorpay_signature: string
}

export async function fetchPaymentConfig(): Promise<PaymentConfigResponse> {
  const { data } = await apiClient.get<PaymentConfigResponse>('/payment/config')
  return data
}

export async function createPremiumOrder(): Promise<PaymentCreateOrderResponse> {
  const { data } = await apiClient.post<PaymentCreateOrderResponse>('/payment/create-order', {})
  return data
}

export async function verifyPremiumPayment(body: PaymentVerifyBody): Promise<{ success: boolean }> {
  const { data } = await apiClient.post<{ success: boolean }>('/payment/verify', body)
  return data
}
